From 9b3c105d1f44d825eebf098bd59c542ad58c3540 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Mon, 19 Jun 2017 18:36:50 +0800
Subject: [PATCH 1961/5242] MLK-15104: ASoC: imx-sii902: add constraint for
 channels

commit  6868c6ec26c6ace7ab186dfda36f033687604724 from
https://source.codeaurora.org/external/imx/linux-imx.git

The maximum channel supported by sii902 is 2, but machine
driver use dummy codec, and there is no constraint list
from codec, so add constraint directly in machine driver.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
(cherry picked from commit 037051e60f3b29fb09aae6f2c97a3532482dfc2a)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-sii902x.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sound/soc/fsl/imx-sii902x.c b/sound/soc/fsl/imx-sii902x.c
index f66ecf5..b905992 100644
--- a/sound/soc/fsl/imx-sii902x.c
+++ b/sound/soc/fsl/imx-sii902x.c
@@ -65,6 +65,11 @@ static int imx_sii902x_startup(struct snd_pcm_substream *substream)
 	if (ret)
 		return ret;
 
+	ret = snd_pcm_hw_constraint_minmax(runtime, SNDRV_PCM_HW_PARAM_CHANNELS,
+							1, 2);
+	if (ret)
+		return ret;
+
 	return 0;
 }
 
-- 
1.7.9.5

